app/views/users/_form.html.haml

```diff
@@ -1,9 +1,11 @@
-= simple_form_for [current_facility, @user] do |f|
+= simple_form_for [current_facility, user] do |f|
   .form-inputs
-    = f.input :first_name
-    = f.input :last_name
-    = f.input :email
+    = f.input :first_name, readonly: local_assigns[:edit] && !user.admin_editable?
```

Comment:
This is a lot of logic to have in a view, obviously. Does this make sense as a helper method, or is there a better way of doing this?